pkg/proxy/apis/config/validation/validation.go

```diff
@@ -67,8 +68,17 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 	allErrs = append(allErrs, validateHostPort(config.MetricsBindAddress, newPath.Child("MetricsBindAddress"))...)
 
 	if config.ClusterCIDR != "" {
-		if _, _, err := net.ParseCIDR(config.ClusterCIDR); err != nil {
-			allErrs = append(allErrs, field.Invalid(newPath.Child("ClusterCIDR"), config.ClusterCIDR, "must be a valid CIDR block (e.g. 10.100.0.0/16)"))
+		if config.FeatureGates[string(kubefeatures.IPv6DualStack)] {
```

Comment:
@aojea I added the feature-gate check here since the kubeproxy config supports it.

Reply:
I also moved this into a separate commit since it touches kube-proxy code.

Reply:
@Arvinderpal @vllry seems you are changing the same file https://github.com/kubernetes/kubernetes/pull/79576 , 
I guess that changes in pkg/proxy/apis/config/validation/validation.go makes more sense in @vllry PR, what do you think?

Reply:
Good point. 
@vllry your PR is a lot larger and my guess is that it will likely take longer to merge into master. Would you be okay with this change being part of a kubeadm PR?
I'm planning a follow up PR for kubeadm changes related to kube-proxy. For example, we'll need to support a slice for `--bind-address` config. If you like, I can move this change to that PR instead.

Reply:
cc @rosti
patching the validation of the kube-proxy of the internal types seems required.
separate commit seems fine by me.

Reply:
the related PR seems to require rebase, so seems fine that we add it as part of this commit, for now.